From 5a46c4941461c2f12d8ba33c0960ce5a2eb55b33 Mon Sep 17 00:00:00 2001
From: Subrata Chatterjee <schatterjee@habana.ai>
Date: Thu, 29 Apr 2021 15:32:05 +0300
Subject: [PATCH] [SW-39285] security: gic: block access from host

Any unprivileged entity (e.g. malicious driver
on host) can exploit the GIC access rights. In other words
keeping GIC accessible from host  makes the system vulnerable
to security threats. So GIC region access is blocked using RR
registers.

With this modification, host can not generate interrupt using
mmGIC_DISTRIBUTOR__5_GICD_SETSPI_NSR anymore. Alternatively
firmware would implement a polling based mechanism to handle
interrupt requests from host.

Change-Id: I166b02058336b6b503884caa91a8e2cddb340427
Signed-off-by: Subrata Chatterjee <schatterjee@habana.ai>
---

diff --git a/gaudi/security.c b/gaudi/security.c
index 45cdc56..ab3d34c 100644
--- a/gaudi/security.c
+++ b/gaudi/security.c
@@ -1262,8 +1262,7 @@
 	LBW_RR_MME1_QM,
 	LBW_RR_MME3_QM_SRAM_CFG,
 	LBW_RR_CPU_TSTAMP,
-	LBW_RR_HBM_GICBGN,
-	LBW_RR_GICEND,
+	LBW_RR_HBM_GIC,
 	LBW_RR_PCIE_DBI_COREBGN,
 	LBW_RR_PCIE_COREEND_AUX_PHY,
 	LBW_RR_PCIE_MSI_PSOC0,
@@ -1303,15 +1302,10 @@
 	max = mmDMA_IF_W_S_BASE;
 	preboot_secure_lbw_rr(LBW_RR_CPU_TSTAMP, min, max);
 
-	/* HBM_CFG up to GIC_L: RR(LBW_RR_HBM_GICBGN) */
+	/* HBM_CFG up to PCIE_WRAP: RR(LBW_RR_HBM_GIC) */
 	min = mmHBM0_BASE - 1;
-	max = HL_CFG_BASE + mmGIC_DISTRIBUTOR__5_GICD_SETSPI_NSR;
-	preboot_secure_lbw_rr(LBW_RR_HBM_GICBGN, min, max);
-
-	/* GIC_H up to PCIE_WRAP: RR(LBW_RR_GICEND) */
-	min = max + sizeof(u32) - 1;
 	max = mmPCIE_WRAP_BASE;
-	preboot_secure_lbw_rr(LBW_RR_GICEND, min, max);
+	preboot_secure_lbw_rr(LBW_RR_HBM_GIC, min, max);
 
 	/*
 	 * Note: we skip here the PCIE_DBI, since iATU might not be configured
